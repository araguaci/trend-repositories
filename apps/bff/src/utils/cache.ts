const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
