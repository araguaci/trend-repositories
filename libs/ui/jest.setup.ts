import '@testing-library/jest-dom';
jest.mock('next/dist/client/router', () => require('next-router-mock'));
