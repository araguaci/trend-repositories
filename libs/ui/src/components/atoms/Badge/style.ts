import styled from 'styled-components';
import Link from '../Link/Link';

export const StyledBadge = styled(Link)`
  display: inline-block;
  box-sizing: border-box;
  border-radius: 2rem;
  padding-left: 10px;
  padding-right: 10px;
  font-size: ${(props) => props.theme.sizes.small};
  background-color: ${(props) => props.theme.colors.blue_3};
  color: ${(props) => props.theme.colors.blue_2};
  line-height: 22px;
  font-weight: ${(props) => props.theme.fontWeight.medium};
  border: 1px solid transparent;

  &:hover {
    background-color: ${(props) => props.theme.colors.blue_1};
    color: ${(props) => props.theme.colors.white_2};
  }
`;
