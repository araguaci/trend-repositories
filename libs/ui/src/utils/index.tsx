/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
