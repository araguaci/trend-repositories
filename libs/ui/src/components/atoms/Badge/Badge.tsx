import { LinkProps } from 'next/link';
import { FC } from 'react';
import { StyledBadge } from './style';

const Badge: FC<LinkProps> = ({ children, href }) => {
  return <StyledBadge href={href}>{children}</StyledBadge>;
};

export default Badge;
