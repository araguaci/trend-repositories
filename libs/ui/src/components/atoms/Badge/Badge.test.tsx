import { renderWithProvider } from '@trend-repositories/ui/utils';
import Badge from './Badge';

describe('Badge Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(
      <Badge href="https://github.com">linguistic</Badge>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
