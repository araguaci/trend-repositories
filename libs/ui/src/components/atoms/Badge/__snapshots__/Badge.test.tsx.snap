// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="style__StyledLink-ikww7b-0 fUma-dV style__StyledBadge-o2pv9p-0 klpobS"
    href="https://github.com"
  >
    linguistic
  </a>
</DocumentFragment>
`;
